src/service/video-manager-impl.js

```diff
@@ -868,12 +926,74 @@ class VideoEntry {
   }
 
   /**
+   * Get the autoplay state distinct from paused by user and paused by
+   * the video scrolling out of view.
+   * @return {boolean}
+   */
+  isAutoplay() {
+    return (this.getPlayingState() === PlayingStates.PLAYING_AUTO) ||
+        (this.playCalledByAutoplay_ && !this.userInteractedWithAutoPlay_);
+  }
+
+  /**
    * Returns whether the video was interacted with or not
    * @return {boolean}
    */
   userInteractedWithAutoPlay() {
     return this.userInteractedWithAutoPlay_;
   }
+
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} opt_vars A map of vars and their values.
+   * @private
+   */
+  analyticsEvent_(eventType, opt_vars) {
+    const trackingVideo = assertTrackingVideo_(this.video);
+    if (trackingVideo) {
+      opt_vars = opt_vars || this.getAnalyticsDetails_(trackingVideo);
+      trackingVideo.element.dispatchCustomEvent(
+          VideoEvents.ANALYTICS, {type: eventType, details: opt_vars});
+    }
+  }
+
+  /**
+   * Collects a snapshot of the current video state for video analytics
+   * @param {!../video-interface.TrackingVideoInterface} video
+   * @return {!../video-interface.VideoAnalyticsDetailsDef}
+   * @private
+   */
+  getAnalyticsDetails_(video) {
+    const autoplay = this.isAutoplay();
```

Comment:
I think this bit will be mostly used to compare performance of autoplay vs normal in experimentation. In that sense, when user interacts with an autoplaying video, we should not change the value of the flag. (we have `video-autoplay-interacted` trigger if they wanna know that)